feat(border): [android] `border-style: none` support

mateoguzmana opened the pull request:
## Summary:

As part of #34425, `borderStyle` value of `none` should be supported.

In this PR, the `BorderStyle` enum is extended and the logic is implemented to make the `none` value work in both `BorderDrawable` and `CSSBackgroundDrawable` as currently there is a feature flag `enableNewBackgroundAndBorderDrawables` controlling which class draws the border. The logic has been implemented in both classes so it works regardless of the feature flag value.

In this PR the border style value doesn't explicitly expose the newly accepted property `none` in the types yet as the implementation for iOS is still needed. Notice the `FlowFixMe` in the examples.

## Changelog:

[ANDROID][ADDED] - `border-style: none` support

## Test Plan:

Two new examples are added. One for `Text` and one for `View` – both of them are setting some properties for the border which by default makes the components draw a border with a default `BorderStyle.SOLID`, but since the property `borderStyle: 'none'` is also passed, that default is overwritten hence you see no border applied.

Both examples have been tested with the `enableNewBackgroundAndBorderDrawables` feature flag on and off.

| Text Example | View Example |
| --------- | ---------- |
| ![Screenshot_1729893684](https://github.com/user-attachments/assets/258e5672-f1ae-465b-a83b-1b22e03196b4) | ![Screenshot_1729893699](https://github.com/user-attachments/assets/5b1c616e-56eb-4073-83e1-b025f34b91bb) |

NickGerleman reviewed (changes requested):
This causes border not to render, which we want, but doesn't account for border widths still being passed to Yoga and consuming layout space (compared to web where it acts as if there is no border present).

mateoguzmana commented:
> This causes border not to render, which we want, but doesn't account for border widths still being passed to Yoga and consuming layout space (compared to web where it acts as if there is no border present).

Oh, that's quite an interesting point. Will look into that, thanks for the very quick feedback! @NickGerleman

mateoguzmana commented:
@NickGerleman I've updated my approach to account for the widths that were still being passed to Yoga and consuming layout space.

There are still a few things I need to tweak in this PR, but before putting more effort into that, I would appreciate some feedback on whether this approach is heading in the right direction. I've also updated my PR description and examples to demonstrate that it is working correctly on Android.

NickGerleman reviewed (changes requested):
Adding `BorderStyle` to Yoga would unfortunately be more involved than what is in this PR. Some of the code here is normally generated, in the Yoga repo (see the copied "generated from enums.py"), and the edits here are currently to private boundary in Yoga we are wanting to move things off of, instead of public API. We also want new code in Yoga to have unit tests (which can be added to OSS repo).

We also don't want to add APIs to Yoga which don't do anything (like, the layout engine doesn't care about whether a border is dotted or solid. It just needs to know the insets the border will take).

I think this would need to be organized as deriving border widths set on Yoga node, from the borderStyle (to know if borders should be ignored). Though we should also be careful not to add too much complexity here, if this change ends up being very involved.

mateoguzmana commented:
Abandoning this PR as could not allocate time to find a simpler way to derive the border
